src/validations/rules/sar.sch

```diff
@@ -175,6 +181,215 @@
+            <sch:let
+                name="matching-control-ids"
+                value="
+                    for $i in $combined
+                    return
+                        $profile-doc//oscal:alter[@control-id = $i]/oscal:add[contains(@by-id, '_obj.')][oscal:prop[@name = 'response-point']]/@by-id" />
+            <!-- targetid values of the findings of the most recent result -->
+            <sch:let
+                name="first-result-finding-target-IDs"
+                value="//oscal:result[@uuid = $first-result]//oscal:finding//oscal:target[@type = 'objective-id']/@target-id" />
+            <!-- matching profile control ids with first result finding target ids -->
+            <sch:let
+                name="target-IDs-combined"
+                value="$matching-control-ids[. = $first-result-finding-target-IDs]" />
+            <!-- profile control ids that are not in the first result finding target ids -->
+            <sch:let
+                name="target-IDs-combined-missing"
+                value="$matching-control-ids[not(. = $first-result-finding-target-IDs)]" />
+            <sch:assert
+                diagnostics="no-duplicate-target-id-diagnostic"
+                doc:guide-reference="Guide to OSCAL-based FedRAMP Security Assessment Plans (SAR) §4.3"
+                id="no-duplicate-target-id"
+                role="error"
+                test="
+                    if (oscal:finding[parent::oscal:result[@uuid = $first-result]])
+                    then
+                        count(oscal:finding/oscal:target[@type = 'objective-id']/@target-id) = count(distinct-values(oscal:finding/oscal:target[@type = 'objective-id']/@target-id))
+                    else
+                        true()">All objective targets in the most recent result assembly must be distinct.</sch:assert>
+        </sch:rule>
+        <sch:rule
+            context="oscal:actor">
+            <sch:let
+                name="SAP-parties"
+                value="$sap-doc/oscal:assessment-plan/oscal:metadata/oscal:party/@uuid" />
+            <sch:let
+                name="SAR-parties"
+                value="/oscal:assessment-results/oscal:metadata/oscal:party/@uuid" />
+            <sch:assert
+                diagnostics="has-matching-SAP-party-diagnostic"
+                doc:guide-reference="Guide to OSCAL-based FedRAMP Security Assessment Plans (SAR) §4.3"
+                id="has-matching-SAP-party"
+                role="error"
+                test="
+                    if (../../oscal:target[@type = 'objective-id'])
+                    then
+                        if (@actor-uuid[. = $SAP-parties])
+                        then
+                            true()
+                        else
+                            if (@actor-uuid[. = $SAR-parties])
+                            then
+                                true()
+                            else
+                                false()
```

Comment:
This simplifies to `(@actor-uuid[. = $SAP-parties]) or (@actor-uuid[. = $SAR-parties])`.